src/SimpleWebRequest.ts

```diff
@@ -802,6 +804,9 @@ export class SimpleWebRequest<TBody, TOptions extends WebRequestOptions = WebReq
                             body = JSON.parse(this._xhr.responseText);
                         } catch (ex) {
                             // If a service returns invalid JSON in a payload, we can end up here - don't crash
+                            // responseParsingFailed flag will indicate that we got response from the server that was corrupted.
+                            // This will be manifested as null on receipient side and flag can help in understanding the problem.
```

Comment:
I like the idea for this, but let's take it all the way through -- can you change it to actually just be "responseParsingException?: Exception" and pass the exception through?  Non-undefined then means there was a problem here.

Reply:
I introduced new interface - `Exception` and a function which maps the JS Error into the `Exception` object, which will be returned to user.

Reply:
Oh!  I'm so sorry.  Been writing too many languages recently and used the wrong type name (Exception instead of Error).  I just meant to pass the raw ex object (which is an Error-type) through there, not to have to process it into the exception interface.  May as well let the consumer of the object read the raw error to display that to the user or present through telemetry any way that's useful.  I'm happy to change that around if you want, or you can do it.

Reply:
No problem. I refactored the code.